litestar/controller.py

```diff
@@ -144,6 +145,11 @@ class Controller:
     """A sequence of dictionaries that to the schema of all route handlers under the controller."""
     signature_namespace: dict[str, Any]
     """A mapping of names to types for use in forward reference resolution during signature modelling."""
+    signature_types: list[Any] | None
```

Comment:
```suggestion
    signature_types: Sequence[Any]
```

Is there a reason to make it optional? It's not optional for `signature_namespace`.

Reply:
I don't think there is a reason for optional here. And `Sequence` should be appropriate too. Also, `signature_namespace` should be `Mapping` as well. Cheers!

Reply:
I resisted the urge to change the `signature_namespace` for now, mostly just to keep this PR focused.